client/src/pages/customer-pages/Home/index.js

```diff
@@ -55,39 +112,101 @@ class Home extends Component {
 
   handleCheck = e => {
     e.preventDefault();
-    const { textInputValueFlightNo } = this.state;
-    console.log(textInputValueFlightNo);
-    const data = JSON.stringify({
-      textInputValueFlightNo
-    });
-    fetch("/api/v1/check_flight", {
-      credentials: "same-origin",
-      headers: {
-        "content-type": "application/json"
-      },
-      method: "POST",
-      body: data
-    })
-      .then(response => response.json(data))
-      .then(data => {
-        sessionStorage.setItem("flight_no", data.flight_no);
-        console.log(data.flight_no);
+    const {
+      textInputValueFlightNo,
+      textInputValuepPhone,
+      checkboxValue
+    } = this.state;
 
-        if (data.isAvailable) {
-          this.setState({
-            isAvailable: true
+    if (checkboxValue === false) {
+      if (textInputValueFlightNo !== "") {
+        const data = JSON.stringify({
+          textInputValueFlightNo
+        });
+        fetch("/api/v1/check_flight", {
+          credentials: "same-origin",
+          headers: {
+            "content-type": "application/json"
+          },
+          method: "POST",
+          body: data
+        })
+          .then(response => response.json(data))
+          .then(data => {
+            sessionStorage.setItem("flight_no", data.flight_no);
```

Comment:
@Balsam-Faysal Good thinking, you went around the problem. But THERE IS A SOLUTION :)
I had to play around a bit with your files to achieve this, and here is the way to do it WITHOUT `sessionStorage`.
First of all, your routes are not the same as your routes when you were using handlebars.
Back in the day, each route was deeply separated from all other routes (except when it came to the cookie and the localstorage). But here in react, routes are similar to directories inside a folder...
Let me clear out: In a folder, you've already got everything set, you only have to open the sub-directory, no load has to be brought, nothing is going to be transferred to your harddrive.
Same goes for your routes: You've already got your bundle.js (your entire website) and you only have to move between routes. You just have to do it right. And here is the way: 

Step 1: **DO NOT** use `window.location` because it breaks the concept I mentioned seconds ago, and separates routes in a way that defies React's logic.
Step 2: You need your state (your `flight_no`) to be stored not inside your route, but inside the parent of Routes. To achieve this I did the following, I modified your `/client/Router.js` as follows:
```jsx
// ... imports
class Router extends Component {
  state = {
    flightNo: "",
    globalProps: null
  };

  ...

  // This function will be triggered by the child route
  // storing your data in here, accessible by all routes. 
  setGlobalProps = props => this.setState({ globalProps: props })

  render() {
    return (
      <BrowserRouter>
        <Switch>
          <Route
            exact
            path="/"
            render={props => (
              <Home
                globalProps={this.state.globalProps}
                setGlobalProps={this.setGlobalProps}
                {...props}
              />
            )}
          />
          <Route
            path="/flightinfo"
            render={() => <FlightInfo globalProps={this.state.globalProps} />}
          />
          ...
        </Switch>
      </BrowserRouter>
    );
  }
}

export default Router;
```
3 things have changed in the previous file, 1 I added the state variable, 2 I made a function that changes that variable, and 3 I rendered `props` and passed it to your `Home` component; Because: 
props are inherited from your `BrowserRouter` which contains a prop called `history`. So you can change the Route using `this.props.history.push('/newRoute');`

Step 3: `Home` component: I replaced your `sessionStorage.setItem` with my `setGlobalProps`.
Step 4: `Home` component: I replaced your `window.location` with `this.props.history` as mentioned earlier
Step 5: `FlightInfo` component: I replaced your `sessionStorage.getItem` with my `this.props.globalProps`
And Valhalla :fireworks: 
You've transferred state between components :)

I hope you think this is a better solution than yours, as it takes more advantage of React's and React-router's abilities.
Note: If the user refreshes the page (in my solution) the inputted data would be erased from the state, meaning, if you don't think it is the user's fault (I do) and feel the responsibility to fix it for the user, then you still can use the session storage to achieve this.